code/modules/mob/living/carbon/life.dm

```diff
@@ -384,3 +383,36 @@
 		if(360.15 to INFINITY) //360.15 is 310.15 + 50, the temperature where you start to feel effects.
 			//We totally need a sweat system cause it totally makes sense...~
 			bodytemperature += min((body_temperature_difference / BODYTEMP_AUTORECOVERY_DIVISOR), -BODYTEMP_AUTORECOVERY_MINIMUM)	//We're dealing with negative numbers
+/////////
+//LIVER//
+/////////
+
+/mob/living/carbon/proc/handle_liver()
+	var/obj/item/organ/liver/liver = getorganslot("liver")
+	if(((!(NOLIVER in dna.species.species_traits)) && (!liver)) || liver.damage >= 100)
+		liver.failing = TRUE
+		liver_failure()
+	else
+		liver.failing = FALSE
+
+/mob/living/carbon/proc/undergoing_liver_failure()
+	var/obj/item/organ/liver/liver = getorganslot("liver")
+	if(liver.failing)
+		return TRUE
+
+/mob/living/carbon/proc/return_liver_damage()
+	var/obj/item/organ/liver/liver = getorganslot("liver")
+	if(liver)
+		return liver.damage
+
+/mob/living/carbon/proc/applyLiverDamage(var/obj/item/organ/liver/L, var/d)

```

Comment:
mob knows its own liver, unless you expect it to have multiple livers and want to damage a specific one